File: python/ejemplos/fibonacci.py
# versión recursiva 'ineficiente'.
def fibonacci(n):
    if n <= 2: return 1
    return fibonacci(n-1) + fibonacci(n-2)

# versión iterativa.
def fib(n):
    f1 = 0
    f2 = 1
    for i in range(n):
        tmp = f2
        f2 = f1 + f2
        f1 = tmp
    return f1

File: R/ejemplos/fibonacci.R
fibonacci <- function(n)
{
##    (1/(sqrt(5))) * ((1 + sqrt(5)) /2)^n - (1/sqrt(5)) * ((1 - sqrt(5))/2)^n
    q <- sqrt(5)
    (1/q * ((1+q)/2)^n - (1/q) * ((1 - q)/2)^n)
}



fib.recursive <- function(n)
{
   if (n == 1 || n == 2) return (1)
   else return ( fib1(n-1) + fib1(n - 2))
}



fib.iterative <- function(n)
{
    if (n == 1 || n == 2) return (1)
    else
    {
        n_1 <- 1
	n_2 <- 1
        for (i in 1:(n-2)) {
	    tmp <- n_2
	    n_2 <- n_2 + n_1
	    n_1 <- tmp
	    }
	return(n_2)
    }
}